lib/miq_expression.rb

```diff
@@ -1256,6 +1256,13 @@ def fields(expression = exp)
     end
   end
 
+  def eql?(other)
```

Comment:
I don't think this is the right method because `eql?` is designed to show two object share the same hash key<sup>[[ref](https://ruby-doc.org/core-2.6.8/Object.html#method-i-eql-3F)]</sup> (i.e. `exp1.hash == exp2.hash`).  I think we only want `==` here.

Either that, or this should implement `==` and `.eql?` should be the alias as per the doc recommendation:

> Subclasses normally continue this tradition by aliasing eql? to their overridden == method

Reply:
That being said, why is this in this PR?  It seems unrelated to setting conditions.

Reply:
I can change the spec.

In the spec it was easier to just use a regular `==` comparison:
https://github.com/ManageIQ/manageiq/pull/21365/files#diff-c7d4c143703284ac0c4c4c467eb951b39dd1e2a0f2291d301149a46a646cff87R996